lambda/src/main/scala/pricemigrationengine/handlers/NotificationEmailHandler.scala

```diff
@@ -39,10 +39,12 @@ object NotificationEmailHandler {
       street <- requiredField(contact.MailingAddress.street, "Contact.MailingAddress.street")
       postalCode <- requiredField(contact.MailingAddress.postalCode, "Contact.MailingAddress.postalCode")
       country <- requiredField(contact.MailingAddress.country, "Contact.MailingAddress.country")
-      newPrice <- requiredField(cohortItem.newPrice.map(_.toString()), "CohortItem.newPrice")
+      estimatedNewPrice <- requiredField(cohortItem.estimatedNewPrice.map(_.toString()), "CohortItem.estimatedNewPrice")
       startDate <- requiredField(cohortItem.startDate.map(_.toString()), "CohortItem.startDate")
       billingPeriod <- requiredField(cohortItem.billingPeriod, "CohortItem.billingPeriod")
 
+      _ <- updateCohortItemStatus(cohortItem.subscriptionName, EmailSendProcessing)
```

Comment:
Instead of a `processing` state, would it be better to have a `complete` and a `failed` state?  If the message fails to go on the queue, could update state to `failed` which indicates it was tried.  

I'm just thinking that if you query the table at any moment to find all the messages that appear to have failed so that you could cross-refer them with what Braze says it sent, the `processing` state might be misleading as they haven't definitely failed at that moment.  I know that's quite an unlikely scenario.

Reply:
There is transaction support in Dynamo although it adds some complexity:
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/transaction-example.html so could read and write back state in a single transaction.

Reply:
Ignore this comment if too complicated.  Solution is fine.

Reply:
I think there are lots of obscure failure cases eg if dynamoDB goes down or is not contactable during email run. 

In which case it wouldnt be possible to set dynamoDB to a 'Failed' state.

Conversely a network problem could cause the SQS send to look like a failure when the email was actually sent, so in that case setting the status to a Failed state would be miss-leading.

In any failure case the action is the same ie to go and reconcile the items still in the 'processing' state with braze/membership-workflow and explisitly set

Reply:
👍